### DISCUSSION TITLE
Integration test

### DISCUSSION THREAD

**Discussion by atari-monk** [bilal-fazlani/commanddotnet#452] [Q&A]
Hi @drewburlingame 
I wrote test
![image](https://user-images.githubusercontent.com/5488398/184666602-1365f764-e84b-4683-942a-f43483e5cc65.png)
In fixture i have AppRunner.
Is there easy way i can get output to console ?
i would like to assert against it in test.
I would like to have that output in debug console, in some property.
Is this doeable ?
I dont see any getter like this in  AppRunner.

**Comment by atari-monk** [comment]
I tried to use testtools to do code as in test section of docs
but i get some exception after using tooltest nuget

**Comment by atari-monk** [comment]
![image](https://user-images.githubusercontent.com/5488398/184676494-6b55436a-b566-4dcc-afdc-828c8d444e9a.png)

**Reply by atari-monk** [reply]
I used prev version of test tools and it works
There is new problem
![image](https://user-images.githubusercontent.com/5488398/184679957-c004f126-41d9-4cc3-89c7-0863cc1c130e.png)
Command seem to be as expected in debug console
but output is empty (assertion passes)
Maybe i use logger and AppRunner doesnt register this output string?

**Reply by atari-monk** [reply]
I dont get why it;s so hard for me to get output

**Reply by drewburlingame** [reply]
Can you paste the code of your command?  The Output will capture whatever is written to the IConsole.  If you're writing directly to System.Console, you'll have to intercept the console as desribed [here](https://commanddotnet.bilal-fazlani.com/gettingstarted/getting-started-400-tests/).  If your output is only to the logger, then you can still get this work if you get your logger to log to the console and intercept the console.

**Reply by atari-monk** [reply]
```
[Theory]
    [MemberData(nameof(ItemReadData.Read01)
        , MemberType= typeof(ItemReadData))]
    public void Test02(int index, string[] cmd, string expected)
    {
        // fixture.Booter.GetAppRunner().Verify(new Scenario
        //     {
        //         When = { Args = string.Join(' ', cmd) },
        //         Then = { 
        //             Output = "[18:36:18 INF] Read Item"
        //                 + "Id | Name | Description | Category | CategoryId |"
        //                 + "10 | Name | Description |          |            |" 
        //         }
        //     });
        fixture.RunCmd(fixture.Booter, cmd);
        var logger = fixture.Booter.GetLogger();
        logger
            .Should()
            .HaveMessage("{0} {1}")
            .Appearing().Once()
            .WithProperty("0")
            .WithValue("Read")
            .And
            .WithProperty("1")
            .WithValue("Item");
        var output = fixture.Booter.GetOut() as IOutMock;
        fixture.AssertItemCount(fixture.Uow, index + 1);
        var item = fixture.GetItem(fixture.Uow, index);
        Assert.Equal(expected.Replace("{id}", item.Id.ToString()), output?.OutText);
    }
```
This works for me.
But i understand at last why Output  was empty.
Becouse i used test condifguration of my app where insted of Console i use some mock.
I forgot.
Sorry.
Tomorow i will use Console and have it working.

**Reply by atari-monk** [reply]
I read quickly, just now, docs on testing, overview and tips for xUnit.
I will stick to my primitive, but fast integration test helpers.
I may get back to this fancy stuff, but right now i need to code some feature and this test is enough to support me in it.
